hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsTestHelper.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.hadoop.fs.azurebfs.services.PrefixMode;
+import org.apache.hadoop.fs.azurebfs.utils.TracingContext;
+
+public class ITestAbfsTestHelper {
+
+  /**
+   * To assert that a path exists as implicit directory we need two things to assert.
+   * 1. List blobs on the path should return some entries.
+   * 2. GetBlobProperties on path should fail.
+   * @param path to be checked
+   * @param fs AzureBlobFileSystem for API calls
+   * @return boolean whether the path exists as Implicit directory or not
+   */
+  public static boolean isImplicit(Path path, AzureBlobFileSystem fs) throws AzureBlobFileSystemException {
+    path = new Path(fs.makeQualified(path).toUri().getPath());
+    if (fs.getAbfsStore().getPrefixMode() == PrefixMode.BLOB) {
+      List<BlobProperty> blobProperties = fs.getAbfsStore()
+        .getListBlobs(path,null, Mockito.mock(TracingContext.class), 2, true);
+      if (blobProperties.size() == 0) {
+        return false;
+      }
+      try {
+        fs.getAbfsStore().getBlobProperty(
+            path,
+            Mockito.mock(TracingContext.class)
+        );
+      }
+      catch (AbfsRestOperationException ex)      {
+        return true;
+      }
+      return false;
+    }
+    else if (fs.getAbfsStore().getPrefixMode() == PrefixMode.DFS) {
+      List<BlobProperty> blobProperties = fs.getAbfsStore()
+          .getListBlobs(path,null, Mockito.mock(TracingContext.class), 2, true);
+      if (blobProperties.size() == 0) {
+        return false;
+      }
+      try {
+        fs.getAbfsStore().getFileStatus(
+            path,
+            Mockito.mock(TracingContext.class)
+        );
+      }
+      catch (Exception ex)      {
+        return true;
+      }
+      return false;
+    }
+    return false;
+  }
+
+  /**
+   * To assert that a path exists as explicit directory
+   * For PrefixMode Blob: GetBlobProperties on path should succeed and marker should be present
+   * For PrefixMode DFS: GetPathStatus on path should succeed and marker should be present
+   * @param path to be checked
+   * @param fs AzureBlobFileSystem for API calls
+   * @return boolean whether the path exists as Implicit directory or not
+   */
+  public static boolean isExplicit(Path path, AzureBlobFileSystem fs) throws Exception{
+    path = new Path(fs.makeQualified(path).toUri().getPath());
+    if (fs.getAbfsStore().getPrefixMode() == PrefixMode.BLOB) {
+      BlobProperty prop;
+      try {
+        prop = fs.getAbfsStore().getBlobProperty(
+            path,
+            Mockito.mock(TracingContext.class)
+        );
+      }
+      catch(AzureBlobFileSystemException ex) {
+        return false;
+      }
+      return prop.getIsDirectory();
+    }
+    else if (fs.getAbfsStore().getPrefixMode() == PrefixMode.DFS) {
+      FileStatus status;
+      try {
+        status = fs.getAbfsStore()
+            .getFileStatus(path, Mockito.mock(TracingContext.class));
+      }
+      catch (Exception ex) {
```

Comment:
Is generic exceprion catch expected ?

Reply:
Here we want to assert that the calls succeed. No matter with what error they fail, we have to return false.
Thoughts??

Reply:
Looks okay to me in case for any error it fails, it is a non-directory.

Reply:
Changed to what thrown by function IOException